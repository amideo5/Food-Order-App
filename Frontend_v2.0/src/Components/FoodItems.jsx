import { useState } from "react";
import { Button } from "react-bootstrap";
import "../assets/transactions.css";

export default function FoodItems({ data, addToCart }, prop) {
  const [order, setOrder] = useState();
  const obj = {};

  function handleChange(e) {
    setOrder(e.target.value);
  }

  function submitOrder() {
    obj["Id"] = data.foodName;
    obj[data.foodName] = order;
    addToCart(obj);
  }

  return (
    <>
      <tr id={data.id}>
        <td>{data.foodName}</td>
        <td>{data.foodPrice}</td>
        <td>{data.foodCategory}</td>
        <td>
          <input
            type="number"
            name="quantity"
            required
            placeholder="Quantity"
            value={order}
            onChange={handleChange}
            
          />
        </td>
        <td>
          <Button onClick={submitOrder}>Add to Cart</Button>
        </td>
      </tr>
    </>
  );
}
