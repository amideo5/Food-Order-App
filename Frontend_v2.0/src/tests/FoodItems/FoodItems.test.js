import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";
import FoodItems from "../../Components/FoodItems";
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Testing Food Items Component", () => {
  const data = {
    id: "1",
    foodName: "Burger",
    foodPrice: 10,
    foodCategory: "Fast Food",
  };

  const addToCart = jest.fn();

  test("Snapshot testing",()=>{
    const {container} = render(<FoodItems data={data} addToCart={addToCart}/>);
    expect(container).toMatchSnapshot();
    });

  it("should render the component with the given props", () => {
    render(<FoodItems data={data} addToCart={addToCart} />);
    expect(screen.getByText("Burger")).toBeInTheDocument();
    expect(screen.getByText("10")).toBeInTheDocument();
    expect(screen.getByText("Fast Food")).toBeInTheDocument();
  });

  it("should update the state on quantity input change", () => {
    render(<FoodItems data={data} addToCart={addToCart} />);
    const quantityInput = screen.getByPlaceholderText("Quantity");
    fireEvent.change(quantityInput, { target: { value: 2 } });
    expect(quantityInput.value).toBe("2");
  });

  it('should call addToCart function on clicking the "Add to Cart" button', () => {
    render(<FoodItems data={data} addToCart={addToCart} />);
    const addToCartButton = screen.getByText("Add to Cart");
    fireEvent.click(addToCartButton);
    expect(addToCart).toHaveBeenCalled();
  });
});
