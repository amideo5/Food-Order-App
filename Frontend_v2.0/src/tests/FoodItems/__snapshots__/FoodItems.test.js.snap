// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Food Items Component Snapshot testing 1`] = `
<div>
  <tr
    id="1"
  >
    <td>
      Burger
    </td>
    <td>
      10
    </td>
    <td>
      Fast Food
    </td>
    <td>
      <input
        name="quantity"
        placeholder="Quantity"
        required=""
        type="number"
        value=""
      />
    </td>
    <td>
      <button
        class="btn btn-primary"
        type="button"
      >
        Add to Cart
      </button>
    </td>
  </tr>
</div>
`;
